lib/event-handler.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const {getProperty, hasProperty, setProperty} = require('./utils.js')
+
+module.exports = {
+  parseEvent
+, processEvent
+}
+
+function parseEvent(event) {
+  if (!hasProperty(event, 'Records')) return undefined
+
+  const records = getProperty(event, 'Records')
+  const checkRecords = Array.isArray(records) && records.length > 0
+  if (!checkRecords) return undefined
+
+  let record = records[0]
+  if (hasProperty(record, 's3.object.key')) {
+    const key = getProperty(record, 's3.object.key').replace(/\+/g, ' ')
+    record = setProperty(record, 's3.object.key', decodeURIComponent(key))
+  }
+
+  return {
+    event: {
+      version: getProperty(record, 'eventVersion')
+    , source: getProperty(record, 'eventSource')
+    , time: getProperty(record, 'eventTime')
+    , name: getProperty(record, 'eventName')
+    }
+  , s3: {
+      region: getProperty(record, 'awsRegion')
+    , user: getProperty(record, 'userIdentity.principalId')
+    , schema: getProperty(record, 's3.s3SchemaVersion')
+    , configuration: getProperty(record, 's3.configurationId')
+    }
+  , bucket: {
+      name: getProperty(record, 's3.bucket.name')
+    , owner: getProperty(record, 's3.bucket.ownerIdentity.principalId')
+    , arn: getProperty(record, 's3.bucket.arn')
+    }
+  , object: getProperty(record, 's3.object')
+  }
+}
+
+function processEvent(eventData) {
```

Comment:
also minor, but both `parseEvent` and `processEvent` are essentially doing the same thing (creating a new, restructured object from an event/eventData). and as far as I can tell, they are always called together. is there any reason to keep them separate?

feels like this could just be a single function/file called `format-event.js` that does this in 1 step

Reply:
combined them in `handleEvent`